server/lib/chat/chat-service.ts

```diff
@@ -589,20 +614,36 @@ export default class ChatService {
             userId: msg.userId,
             time: Number(msg.sent),
           })
-          users.push({ id: msg.userId, name: msg.userName })
+          userIds.add(msg.userId)
           break
 
         default:
           return assertUnreachable(msg.data)
       }
     }
 
-    const mentions = await findUsersById(Array.from(mentionIds))
+    const [users, userMentions, channelMentions] = await Promise.all([
+      findUsersById(Array.from(userIds)),
+      findUsersById(Array.from(userMentionIds)),
+      this.getChannelInfos(Array.from(channelMentionIds), userId),
```

Comment:
imo this should return the same data as if the messages were sent 1-by-1, or you're opening the door to weird bugs that will be hard to track down and potentially making this hard to cache in the future (that is, this should NOT include user-specific information)